internal/pkg/describe/env.go

```diff
@@ -146,6 +150,28 @@ func (d *EnvDescriber) Version() (string, error) {
 	return metadata.Version, nil
 }
 
+// ServiceDiscoveryEndpoint returns the endpoint the environment was initialized with, if any. Otherwise,
+// it returns the legacy app.local endpoint.
+func (d *EnvDescriber) ServiceDiscoveryEndpoint() (string, error) {
+	p, err := d.Params()
+	if err != nil {
+		return "", fmt.Errorf("get params of environment %s in app %s: %w", d.env.Name, d.env.App, err)
+	}
+	for k, v := range p {
+		// Ignore non-svc discovery params
+		if k != cfnstack.EnvParamServiceDiscoveryEndpoint {
+			continue
+		}
+		// If the service discovery param is set (meaning the stack is not upgraded), use the provided value
+		if v != "" {
+			return v, nil
+		}
+		// if the param exists but is empty, use the legacy endpoint.
```

Comment:
When can this happen? (not against the code or anything) but it might be worthwhile writing it.

My understanding is that when upgrading we always set a value for this parameter.

Reply:
We don't set a value when upgrading. If the stack is created at version 1.5.0 or higher, it's set to `env.app.local`. If the stack is upgraded from before that version, it's set to empty, which causes the template to use `app.local`.